backend/src/nodes/nodes/image_utility/inpaint.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+from enum import Enum
+
+import cv2
+import numpy as np
+
+from . import category as ImageUtilityCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, NumberInput, EnumInput
+from ...properties.outputs import ImageOutput
+
+
+class InpaintAlgorithm(Enum):
+    NS = cv2.INPAINT_NS
+    TELEA = cv2.INPAINT_TELEA
+
+
+@NodeFactory.register("chainner:image:inpaint")
+class InpaintNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Inpaint Image"
+        self.inputs = [
+            ImageInput(channels=[1, 3]),
+            ImageInput(label="Mask", channels=1),
+            EnumInput(
+                InpaintAlgorithm,
+                option_labels={
+                    InpaintAlgorithm.NS: "Navier Stokes",
+                    InpaintAlgorithm.TELEA: "Telea",
+                },
+            ),
+            NumberInput(
+                "Search Radius",
+                minimum=0,
+                default=1,
+                precision=1,
+                controls_step=1,
+            ),
+        ]
+        self.outputs = [ImageOutput(image_type="Input0")]
```

Comment:
Time for some more type system tricks!

Firstly, since we know that the image and its mask must have the same size, we can use a type intersection to model this:
```
Image {
    width: Input0.width & Input1.width,
    height: Input0.height & Input1.height,
    channels: Input0.channels
}
```

This has a nice side effect: we know the size of the output image, even if we only know the size of one of the input images. E.g. if the size the main image is unknow, but the mask is 100x100, then we know that the node will either output a 100x100 image or error.

However, this creates an interesting case: If we know the sizes of both images and they are different, then we statically know the node will always error.
In terms of the type system, this is represented by the output type being `never`. `never` is the same concept as TypeScript's and Python's `never` type, it's the type that represents the empty set. When an output is `never`, we know that an error is going to occur, but we don't know that that error is. So outputs can optionally have a never reason. This a simple error message that will be shown to the user if an output is never.

With this little trick, we can statically verify that the 2 images have the same size using the type system. The [Add Normals node](https://github.com/chaiNNer-org/chaiNNer/blob/main/backend/src/nodes/nodes/image_filter/normal_addition.py#L42) is doing the same thing btw.

Reply:
So, this helps to raise an error before python execution? Interesting! 
Thanks for detailed explanation.